src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -771,6 +751,38 @@ static int add_root_mount(void) {
 #endif
 }
 
+static bool esp_is_valid(DissectedPartition *p, sd_id128_t xbootldr_uuid) {
+        sd_id128_t loader_uuid;
+        int r;
+
+        if (!is_efi_boot()) {
+                log_debug("Not an EFI boot, skipping ESP check.");
+                return true;
+        }
+
+        r = efi_loader_get_device_part_uuid(&loader_uuid);
+        if (r < 0) {
+                /* Won't be able to check if this is the boot ESP, thus not mounting it */
+                if (r == -ENOENT)
+                        log_debug("EFI loader partition unknown, not mounting ESP.");
```

Comment:
please still use log_debug_errno() here, and pass `r` here, even if you don't use `%m` in the format string. the error number is still included in the structure log message as a separate field then, which is a good thing.

i.e. log_xyz_errno() should be used primarily when there's an error code we are acting on, and only secondarily to make %m output something nice